### PR TITLE
Expose native FP32 AdamW for controlled retrieval fine-tuning

### PR DESCRIPTION
**Pull request by shan-nvidia**
## What

Stacked on #353 (which is stacked on #351). Make the experimental native
FP32 AdamW option explicit in public recipe code, rather than a local overlay.
The protocol and optimizer changes are separate PRs and commits.

## Why

The controlled retrieval experiments use native FP32 AdamW, while the public
recipe defaults select FusedAdam or FlashAdamW. Reproducible comparisons require
an explicit configuration, not silent changes in installed packages.

## Usage

```yaml
optimizer_backend: torch_adamw
```

This opts into **FP32 model parameters, compute, reductions, and Adam moments**
with fused native `torch.optim.AdamW`. It consumes more memory and may run slower
than BF16/quantized FlashAdamW. It is not just FP32 checkpoint serialization or
BF16 compute with FP32 master weights. Existing `auto` defaults are unchanged.

## How

Promote the previously tested isolated training overlay without changing its
semantics: explicit native optimizer, FP32 FSDP policy, post-setup parameter
dtype enforcement, and rank-zero optimizer-checkpoint moment verification.
Skip imports of unused optional optimizer extensions for explicit backends.

## Validation

Regression tests cover text/VL native configuration, unchanged optional
optimizer selection, and rejection of missing/non-FP32 checkpoint moments.
The identical train.py overlay has already completed the bounded eight-GPU
Draco pilots (including checkpoint/resume) and four batch/LR sweep arms;
each verified 368 FP32 exp_avg and 368 FP32 exp_avg_sq tensors.
No new GPU run or metric improvement is claimed for publishing this patch.
`MISTRAL3_VL_EMBED_MODEL=fixture/model python -m pytest
tests/recipes/embed/test_finetune_native_fp32.py
tests/recipes/embed/test_finetune_optimizer_backends.py
tests/recipes/embed/test_finetune_fixed_steps.py -q`: **34 passed**.
The training source SHA256 is unchanged from the validated experimental overlay:
`6628d68f910b36cbdcb4507513931b4bf232f1357139c8a55397affa969580aa`.

### CODE REVIEW

**Comment by shan-nvidia** [comment]
Closing per the consolidated product design. Keep the existing BF16 model/export with FP32 optimizer master weights and moments; full-FP32 model training is not required. Revised #353 documents the contract and #356–358 are restacked without this PR. Existing experiment commit snapshots remain preserved; no running experiment is changed.
